girder/api/v1/resource.py

```diff
@@ -139,6 +140,99 @@ def _getResourceModel(self, kind, funcName=None):
             raise RestException('Invalid resources format.')
         return model
 
+    def _lookUpToken(self, token, parentType, parent):
+        """
+        Find a particular child resource by name or throw an exception.
+        :param token: the name of the child resource to find
+        :param parentType: the type of the parent to search
+        :param parent: the parent resource
+        :returns: the child resource
+        """
+
+        seekFolder = (parentType in ('user', 'collection', 'folder'))
+        seekItem = (parentType == 'folder')
+        seekFile = (parentType == 'item')
+
+        # (model name, mask, search filter)
+        searchTable = (
+            ('folder', seekFolder, {'name': token,
+                                    'parentId': parent['_id'],
+                                    'parentCollection': parentType}),
+            ('item', seekItem, {'name': token, 'folderId': parent['_id']}),
+            ('file', seekFile, {'name': token, 'itemId': parent['_id']}),
+        )
+
+        for candidateModel, mask, filterObject in searchTable:
+            if not mask:
+                continue
+
+            candidateChild = self.model(candidateModel).findOne(filterObject)
+            if candidateChild is not None:
+                return candidateChild, candidateModel
+
+        # if no folder, item, or file matches, give up
+        raise RestException('Child resource not found: {}({})->{}'.format(
+            parentType, parent.get('name', parent.get('_id')), token))
+
+    def _lookUpPath(self, path, user):
+        """
+        Find a particular resource by path or throw an exception.  The given
+        user must have read access to all resources featured in the path.
+        :param path: the path of the object to find
+        :param user: the user to search as
+        :returns: the resource.
+        """
```

Comment:
I think sphinx will throw warnings about this docstring due to not having a blank line after the description. Feel free to remove the docstring altogether; most of our REST route handlers don't have them since they are redundant with the `description` property.